### ISSUE TITLE
So much love to your code!

### ISSUE THREAD

**Issue by faddat** [ahmetalpbalkan/wagl#1]
Ahmet,

I was confused about only one thing here:  

If I use BIND and wagl together, can I publish my DNS records-- that is, make it so that they forward into the global DNS system?

Thank you Ahmet!

**Comment by ahmetb** [comment]
@faddat hello, I haven't used BIND before but I believe you can use wagl for external querying in two ways:
- Direct external recursion with wagl (this works only for simple type of records and has a limited feature set): For this, please look at command line reference (`--external`/`--ns` options). This feature is actually enabled by default on wagl.
- Combining with BIND: You can find a sample example of this done for mesos-dns here: https://mesosphere.github.io/mesos-dns/docs/tutorial-forward.html Just replace `mesos.` domain with `swarm.` and pass `--external=false` to wagl.

Does this help?

**Comment by faddat** [comment]
I am not sure if it will help but in a few hours I'll be finding out :)!

Here's the use-case in detail:

I run a site, klouds.org (github.com/superordinate/klouds) which hosts dockerized open source application containers and then automagically (opt-out, not opt-in) donates a third of the proceeds from the hosting to the author / group behind the application (once we're finished adding support for payments, that is).  I came up with the idea by trying to find a way to use OSS to do something that can't be done with closed software.  

Anyway I am refactoring the stack at the moment, away from an inelegant, but effective cobbling together of mesos, marathon, and consul and towards a 100% docker-based platform using docker swarm + compose + (something to handle the storage but I've not yet figured out the best way to do this, but I will :)

So this leaves me with a whole bunch of subdomains, unless users buy/bring their own domains.  The trickiest part of building all this was the service discovery + virtual host + reverse proxy.  If I can get BIND and wagl to sex properly, I could be my own DNS provider and would no longer need to rely on the *.klouds.org I have set up now, instead just publishing to DNS natively.  It may also free me from the RP (but not the virtual host) as subdomains could resolve to the server the application container is running on.  

Did all that make sense?
